### ISSUE TITLE
Support Problem Details Responses

### ISSUE THREAD

**Issue by jamescrosswell** [getsentry/sentry-dotnet#2523]
### Problem Statement

[.NET 8 is (likely) introducing official support for problem details responses](https://timdeschryver.dev/blog/translating-exceptions-into-problem-details-responses#problem-details-with-aspnet)... but these are also used in the wild already. 

Currently the SentryHttpMessageHandler doesn't pay any particular attention to the response details when an error is encountered. Potentially the Problem Details Response has important information to help users debug issues when Http Requests that they make result in an error however.

**Comment by bitsandfoxes** [comment]
With .NET 9 looming on the horizon we might want to revisit this?

**Comment by jamescrosswell** [comment]
> With .NET 9 looming on the horizon we might want to revisit this?

Yeah, I'm not sure exactly what the implementation would look like. One option would be to store the problem details (basically some JSON) in the Event.Extras... although I think there's a plan to get rid of Extras right? Is there anywhere else we could store this?

**Comment by bitsandfoxes** [comment]
Adding this to the 5.0 milestone either to add support or to have a better understanding how to support this.
With the move to support spans-only there might be a new place to store this. I'll know more once that plan is a bit more finalized (early next week)

**Comment by jamescrosswell** [comment]
I'm not sure if there's anything valuable in these tbh... I can't see examples of anything we don't already capture via other means. 

Leaving this open with a low priority - if you want/need something from Problem Details to be captured bi Sentry, please add some context here and we can consider reprioritising.
